products_application/src/App.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react';
+import styles from './App.module.scss';
+import { Cards } from './components/Cards/Cards';
+import { Loader } from './components/Loader/Loader';
+import { ErrorButton } from './components/ErrorButton/ErrorButton';
+import { IAppState, IRequestResult } from './types';
+
+export class App extends Component<object, IAppState, string> {
+  state = {
+    inputValue: this.setInputValue(),
+    cards: [],
+    isLoading: true,
+    isNothingFound: false,
+    isButtonDisabled: false,
+  };
+
+  setInputValue() {
+    const keyWord = localStorage.getItem('alyona8891_keyword');
+    if (keyWord) {
+      return keyWord;
+    } else {
+      return '';
+    }
+  }
+
+  componentDidMount(): void {
+    const keyWord = localStorage.getItem('alyona8891_keyword');
+    if (keyWord) {
+      this.setState({
+        isButtonDisabled: true,
+      });
+      fetch(`https://dummyjson.com/products/search?q=${keyWord}`)
+        .then((res) => res.json())
+        .then(
+          (json: IRequestResult) => {
+            if (json.products.length === 0) {
+              this.setState({
+                isLoading: false,
+                cards: json.products,
+                isNothingFound: true,
+                isButtonDisabled: false,
+              });
+            } else {
+              this.setState({
+                isLoading: false,
+                cards: json.products,
+                isButtonDisabled: false,
+              });
+            }
+          },
+          (error: Error) => {
+            this.setState({
+              isButtonDisabled: true,
+            });
+            console.log(error);
+          }
+        );
+    } else {
+      this.setState({
+        isButtonDisabled: true,
+      });
+      fetch('https://dummyjson.com/products')
+        .then((res) => res.json())
+        .then(
+          (json: IRequestResult) => {
+            if (json.products.length === 0) {
+              this.setState({
+                isLoading: false,
+                cards: json.products,
+                isNothingFound: true,
+                isButtonDisabled: false,
+              });
+            } else {
+              this.setState({
+                isLoading: false,
+                cards: json.products,
+                isButtonDisabled: false,
+              });
+            }
+          },
+          (error: Error) => {
+            this.setState({
+              isButtonDisabled: true,
+            });
+            console.log(error);
+          }
+        );
+    }
+  }
+
+  handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target) {
+      this.setState({
+        inputValue: e.target.value,
+      });
+    }
+  };
+
+  handleSearchButton = () => {
```

Comment:
as your button is type 'submit', you should use preventDefault, 'cause after submit you can notice this 
![image](https://github.com/Alyona8891/products-application/assets/89910618/6d60ac17-94d7-48d6-8a35-763a0e82475c)

Reply:
redirection happens